### PR TITLE
Bug fixes for DatePicker and Calendar

### PR DESCRIPTION
**Pull request by devongovett**
The individual commits may be easier to review. This fixes many bugs found during testing:

* Day names are not centered (DG)
* If you switch to Japanese calendar it repeats monday twice (DG)
* Dark mode strike outs on unavailable dates are hard to see (RS)
* Remove hover and active state in read only (DG)
* If I click outside the circle but inside the cell, it will start and end my new range immediately, making it feel broken (RS)
* Option+Up arrow doesn't open the popover, only Option+Down arrow. Given that menus, pickers, etc open for both, I feel like this should too (RS)
* For the zoned time story it announces the range as march and not february. Devon says it should be march and is display february. Screenshot row 11 (KT)
* Minor issue with the "ZonedDateTime" text appearing above the popover in the "Events" section of the useDateRangePicker docs. See screenshots row 12 (DL)
* field and datepicker each have their own group, and both of them are announced on focus. (DG)
* VO iphone default TimeField story, change time to 12:20 AM, leave field, come back, tap on AM and change to PM. VO only reads "12 PM" instead of "12:20 PM" (RS)

### CODE REVIEW

**Review by ktabors — DISMISSED** [review]
All of these changes look good.

File: packages/@adobe/spectrum-css-temp/components/calendar/index.css [hunk 1]
```diff
@@ -207,6 +207,7 @@ governing permissions and limitations under the License.
         height: 2px;
         transform: rotate(-16deg);
         border-radius: 1px;
+        background: currentColor;
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@adobe/spectrum-css-temp/components/calendar/index.css:210]: It's the same as the current text color, i.e. the `color` property or an inherited value.

File: packages/@react-aria/calendar/src/utils.ts [hunk 2]
```diff
@@ -50,12 +50,14 @@ export function useVisibleRangeDescription(startDate: CalendarDate, endDate: Cal
...
-    calendar: startDate.calendar.identifier
+    calendar: startDate.calendar.identifier,
+    timeZone
   });
 
   let dateFormatter = useDateFormatter({
     dateStyle: 'long',
-    calendar: startDate.calendar.identifier
+    calendar: startDate.calendar.identifier,
+    timeZone
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-aria/calendar/src/utils.ts:60]: Yeah

[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-aria/calendar/src/utils.ts:60]: I noticed the following when testing voiceover: 
image

The Monday cell is labeled as Sunday for some reason. Looking into it more deeply, it seems like the start and end date are labeled one day behind, perhaps due to time zone reasons. Changing the time doesn't seem to have an effect. Note this was pre-existing so not a blocker for this particular pull

Stories is https://reactspectrum.blob.core.windows.net/reactspectrum/43693ab05f01ed1e3c2a5d63ff23478a9710dbc8/storybook/index.html?path=/story/date-and-time-daterangepicker--defaultvalue-zoned

File: packages/@adobe/spectrum-css-temp/components/calendar/index.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (ktabors) [reply packages/@adobe/spectrum-css-temp/components/calendar/index.css:210]: I assumed it matches the "parent" text color and which changes depending on cell state, but I wasn't seeing where that color was being set. Also, I wasn't sure why we needed to be explicate about the color here.

**Review by LFDanLu — APPROVED** [review]
LGTM, the thing I found can be handled in followup

File: packages/@react-aria/calendar/src/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-aria/calendar/src/utils.ts:60]: oh great catch! I might have to refactor the i18n messages. not sure if I can pass a timezone through to that...

**Review by dannify — APPROVED** [review]

**Pull request merged**